components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/EventHookHandlerServiceComponent.java

```diff
@@ -0,0 +1,160 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.webhook.common.event.handler.internal;
+
+import org.wso2.carbon.identity.event.IdentityEventConfigBuilder;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.IdentityEventServerException;
+import org.wso2.carbon.identity.event.bean.ModuleConfiguration;
+import org.wso2.identity.webhook.common.event.handler.LoginEventHookHandler;
+import org.wso2.identity.webhook.common.event.handler.builder.LoginEventPayloadBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.identity.event.common.publisher.EventPublisherService;
+import org.wso2.identity.webhook.common.event.handler.constant.Constants;
+import org.wso2.identity.webhook.common.event.handler.util.EventHookHandlerUtils;
+
+/**
+ * WSO2 Event Handler service component class.
+ */
+@Component(
+        name = "org.wso2.identity.webhook.common.event.handler.internal.EventHookHandlerServiceComponent",
+        immediate = true)
+public class EventHookHandlerServiceComponent {
+
+    private static final Log log = LogFactory.getLog(EventHookHandlerServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            log.debug("Event Handler is activated.");
+            String isLoginEventHandlerEnabled = getIdentityEventProperty(Constants.LOGIN_EVENT_HOOK_NAME,
+                    Constants.LOGIN_EVENT_HOOK_ENABLED);
+            BundleContext bundleContext = context.getBundleContext();
+
+            if (isLoginEventHandlerEnabled != null && isLoginEventHandlerEnabled.equalsIgnoreCase(Boolean.TRUE.toString())) {
+                bundleContext.registerService(AbstractEventHandler.class.getName(),
+                        new LoginEventHookHandler(EventHookHandlerUtils.getInstance()), null);
+            } else if (isLoginEventHandlerEnabled == null) {
+                log.error("Login Event Handler enabled property is null.");
+            } else {
+                log.error("Login Event Handler is not enabled.");
+            }
```

Comment:
Do we need both. IMO, it's not quite important to log that the "isLoginEventHandlerEnabled" property is null

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/39adc5d3ada05a48388b42bb7303c9c2498322d2